wire-gson-support/src/main/java/com/squareup/wire/WireTypeAdapterFactory.kt

```diff
@@ -48,6 +49,7 @@ import com.squareup.wire.internal.createRuntimeMessageAdapter
 class WireTypeAdapterFactory @JvmOverloads constructor(
   private val typeUrlToAdapter: Map<String, ProtoAdapter<*>> = mapOf(),
   private val writeIdentityValues: Boolean = false,
+  private val loader: Optional<ClassLoader> = Optional.empty()
```

Comment:
You said we cannot use `ClassLoader?`, what's the special meaning for it?

Reply:
That'd be really nice if we could not use `Optional` and pass `ClassLoader?` directly

Reply:
A `null` classloader has special meaning on the JVM. It means "use the boot classloader" (which basically contains the java.lang package). If that's not a use-case you want to support, we can drop the `Optional`, forbid the usage of `null` (except in the two cases like this), and away we go! I was just trying to be maximally semantically correct.